mantidproject/mantid
# Add option to change workspace names

**Description of work.**
This PR changes the individual output workspace names of `ReflectometrySliceEventWorkspace` so they are more descriptive of the slice.
The change was done in the algorithm `FilterEvents` as an optional property.

**To test:**
Run the following script. It should output two groups called `time_sliced` and `log_sliced`. 

In `time_sliced` there should be 3 workspaces called:
- `time_sliced_0_305` 
- `time_sliced_305_610`
- `time_sliced_610_610`
 
In `log_sliced` there should be 3 workspaces called
 - `log_sliced_Log.proton_charge.From.0.001085.To.0.001095.Value-change-direction:both`
 - `log_sliced_Log.proton_charge.From.0.001095.To.0.001105.Value-change-direction:both` 
 - `log_sliced_Log.proton_charge.From.0.001105.To.0.001115.Value-change-direction:both`
```
LoadEventNexus(Filename='INTER38415',OutputWorkspace='TOF_38415',LoadMonitors=True)
input_ws = mtd['TOF_38415']
monitor_ws= mtd['TOF_38415_monitors']

# Slice by time
time_sliced = ReflectometrySliceEventWorkspace(InputWorkspace=input_ws, MonitorWorkspace=monitor_ws, TimeInterval=305)

# Slice by log value
log_sliced = ReflectometrySliceEventWorkspace(InputWorkspace=input_ws, MonitorWorkspace=monitor_ws, LogName='proton_charge',MinimumLogValue=0.00109, MaximumLogValue=0.00111,LogValueInterval=0.00001)
```



Fixes #25535

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
